clang/test/Driver/sycl-obj-remove.cpp

```diff
@@ -0,0 +1,16 @@
+/// Verify object removal when the offload compilation fails.
+
+// REQUIRES: system-linux
+
+// RUN: touch %t.o
+// RUN: not %clangxx -fsycl -Xsycl-target-frontend -DCOMPILE_HOST_FAIL=1 -o %t.o %s
```

Comment:
I'm not sure I understand this test. Could you please elaborate?
I think you are creating a new object file.
And triggering a host compilation failure.
And the expectation is that the object file created gets removed?

The test description mentions, `Verify object removal when the offload compilation fails` , but you are triggering a host compilation failure.
Can you trigger any offload compilation action failure instead?

Reply:
I will update the macro name - it is a device compilation that I'm triggering the failure - the name carried over while I was experimenting with failure modes.

Yes, the expectation is for any pre-existing object file to be removed if the compilation that would potentially overwrite that file fails.  This is standard behavior with compilations creating objects.